components/eventing-controller/pkg/handlers/nats_test.go

```diff
@@ -617,6 +614,287 @@ func TestNatsSubAfterSync_FiltersChange(t *testing.T) {
+		// set metadata on nats subscription
+		if err := natsBackend.subscriptions[key].SetPendingLimits(msgLimit, bytesLimit); err != nil {
+			t.Fatalf("set pending limits for nats subscription failed: %v", err)
+		}
+	}
+
+	// Now, add a new filter to subscription
+	newFilter := sub.Spec.Filter.Filters[0].DeepCopy()
+	newFilter.EventType.Value = fmt.Sprintf("%snew1", eventingtesting.OrderCreatedEventTypeNotClean)
+	sub.Spec.Filter.Filters = append(sub.Spec.Filter.Filters, newFilter)
+
+	// get new cleaned subject
+	secondSubject, err := getCleanSubject(newFilter, cleaner)
+	g.Expect(err).ShouldNot(HaveOccurred())
+	g.Expect(secondSubject).To(Not(BeEmpty()))
+
+	// Sync the subscription
+	_, err = natsBackend.SyncSubscription(sub, cleaner)
+	if err != nil {
+		t.Fatalf("sync subscription failed: %v", err)
+	}
+
+	// Check if total existing NATS subscriptions are correct
+	// Because we have two filters (i.e. two subjects)
+	expectedTotalNatsSubs := 2 * defaultSubsConfig.MaxInFlightMessages
+	g.Expect(natsBackend.subscriptions).To(HaveLen(expectedTotalNatsSubs))
+
+	// Verify that the nats subscriptions for first subject was not re-created
+	for i := 0; i < defaultSubsConfig.MaxInFlightMessages; i++ {
+		natsSub := natsBackend.subscriptions[createKey(sub, firstSubject, i)]
+		g.Expect(natsSub).To(Not(BeNil()))
+		g.Expect(natsSub.IsValid()).To(BeTrue())
+
+		// check the metadata, if they are same then it means that nats subscriptions
+		// were not re-created by SyncSubscription method
+		subMsgLimit, subBytesLimit, err := natsSub.PendingLimits()
+		g.Expect(err).ShouldNot(HaveOccurred())
+		g.Expect(subMsgLimit).To(Equal(msgLimit))
+		g.Expect(subBytesLimit).To(Equal(msgLimit))
+	}
+
+	// Test if subscription is working for both subjects
+	// Send an event on first subject
+	data := fmt.Sprintf("data-%s", time.Now().Format(time.RFC850))
+	expectedDataInStore := fmt.Sprintf("\"%s\"", data)
+	err = SendEventToNATS(natsBackend, data)
+	if err != nil {
+		t.Fatalf("publish event failed: %v", err)
+	}
+	// The sink should receive event for first subject
+	err = subscriber.CheckEvent(expectedDataInStore, subscriberCheckURL)
+	if err != nil && !strings.Contains(err.Error(), "check event after retries failed") {
+		t.Fatalf("check event failed for first subject: %v", err)
+	}
+
+	// Now, send an event on second subject
+	data = fmt.Sprintf("data-%s", time.Now().Format(time.RFC850))
+	expectedDataInStore = fmt.Sprintf("\"%s\"", data)
+	err = SendEventToNATSOnEventType(natsBackend, secondSubject, data)
+	if err != nil {
+		t.Fatalf("publish event failed: %v", err)
+	}
+	// The sink should receive the event for second subject
+	err = subscriber.CheckEvent(expectedDataInStore, subscriberCheckURL)
+	if err != nil {
+		t.Fatalf("check event failed for second subject: %v", err)
+	}
+}
+
+// TestNatsSubAfterSync_FilterRemoved tests the SyncSubscription method
+// when a filter is removed from subscription
+func TestNatsSubAfterSync_FilterRemoved(t *testing.T) {
```

Comment:
can we have a test with multiple subscriptions? 
1 sub will not be modified
1 sub will be modified
both should work as expected before and after modification

if this test already exists, please ignore, i could not find it

Reply:
done